server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -542,25 +552,24 @@ private static void parseArray(ParseContext context, ObjectMapper parentMapper,
             ObjectMapper.Dynamic dynamic = dynamicOrDefault(parentMapper, context);
             if (dynamic == ObjectMapper.Dynamic.STRICT) {
                 throw new StrictDynamicMappingException(parentMapper.fullPath(), arrayFieldName);
-            } else if (dynamic == ObjectMapper.Dynamic.TRUE) {
-                Mapper.Builder builder = context.root().findTemplateBuilder(context, arrayFieldName, XContentFieldType.OBJECT);
-                if (builder == null) {
+            } else if (dynamic == ObjectMapper.Dynamic.FALSE)  {
+                // TODO: shouldn't this skip, not parse?
```

Comment:
Leftover?

Reply:
no this is only moved, it is one of those TODOs that have been there forever. I was planning on leaving it